### DISCUSSION TITLE
Cant connect to docker host

### DISCUSSION THREAD

**Discussion by bwired-nl** [coder/code-server#2016] [Q&A]
HI all,
Im running a docker package on my synology,
im also running code-server as a docker on the synology.
everything works fine, a can create run docker etc. via the terminal within code-server.

The only problem i have that i cannot use the docker extension in code server. it tells i cant connect to the docker daemon on my synology. i tried the settings.json etc, but no luck
keeps telling Docker is not installed...

hope you can give me a hint :-)
thanks all!

**Comment by code-asher** [comment]
Is that the full error ("can't connect to the docker daemon") or is there more?

This is kind of a shot in the dark but if code-server can't find docker while the terminal can then maybe it has to do with the `PATH`? It could be that docker isn't in the `PATH` provided to code-server. 

You could check the location of docker with `command -v docker` or `which docker` then use `ps eww <code-server pid>` to check what its `PATH` is.

**Comment by bwired-nl** [comment]
```
> _peter@DS3617:~$ ps eww 15731
>   PID TTY      STAT   TIME COMMAND
> 15731 ?        Ssl    0:39 containerd --config /var/run/docker/containerd/containerd.toml --log-level info
> peter@DS3617:~$ which docker
> /usr/local/bin/docker
> peter@DS3617:~$ docker -v
> Docker version 18.09.8, build bfed4f5
> peter@DS3617:~$ 
> 
> I see the difference, not sure how to solve this...
> Thanks!!!!_
> 
```

**Reply by code-asher** [reply]
Is that the code-server pid? I would expect it to say `code-server` rather than `containerd`.

Are you able to edit the dockerfile? One idea would be to add `/usr/local/bin` to the path there. For example with:

```
ENV PATH=/usr/local/bin:$PATH
```

**Reply by code-asher** [reply]
Or it might work to set `PATH` using `docker run -e PATH=$PATH [other args]`

**Comment by lotus647** [comment]
Hi code-asher, I hope you are doing well.

Based on the information you provided, it seems that you are encountering an issue with the version of Docker installed on your system.

The output of the docker -v command shows that you have Docker version 18.09.8 installed. However, the containerd process you mentioned is using a different configuration file located at /var/run/docker/containerd/containerd.toml.

To resolve this issue, you have a few options:

Update Docker: Consider updating Docker to a version that is compatible with your current setup. Since your Docker version is relatively old, updating it to a more recent version might resolve the compatibility issue. You can check the Docker website or official documentation for instructions on how to update Docker on your specific operating system.

Use the appropriate configuration file: If you cannot update Docker or prefer to stick with the current version, you can modify the containerd configuration file to use the correct configuration. You can edit the /var/run/docker/containerd/containerd.toml file and ensure that it is configured properly for your Docker version.

Uninstall and reinstall Docker: If updating or modifying the configuration file doesn't resolve the issue, you might consider uninstalling Docker completely and then reinstalling it. This can help ensure a clean installation and potentially resolve any configuration conflicts.

Remember to backup any important Docker data or containers before making any changes to avoid data loss.

If none of these solutions work or if you need further assistance, please provide more details about the specific issue you are facing, including any error messages or logs you may have encountered.

**Reply by lotus647** [reply]
Sorry that was a mouthful

**Reply by code-asher** [reply]
I am not the original poster but I am not sure it is a configuration issue.  If I understand correctly they launch code-server inside Docker (which works) but then the Docker extension inside code-server is not able to connect to Docker.

Actually maybe that just means they need to mount the Docker socket into the container so the extension can use it?

**Reply by code-asher** [reply]
And they probably need to install Docker inside the container if they have not already done so.

**Reply by lotus647** [reply]
Quite possible. I just briefly analyzed the error and then wrote the mouthful, so mine is not very complex.
